### DISCUSSION TITLE
Simple howto on upgrading openmage

### DISCUSSION THREAD

**Discussion by MrJarlsburg** [OpenMage/magento-lts#2188] [Questions & Answers] [answered]
As a user not too familiar with linux, and thus obviously lacing in some basics, I am having a tough time getting openmage updated.
There are howtos on how to migrate magento to openmage, so this wasn't an issue. However I can not find a single thread on how to upgrade from 19.4.8 to ie 19.4.16.
After migrating, the migrate script left a composer.json, so I assume using composer is the way to go.
If I run composer update it updates/installs some dependancies, but i'm still stuck on the same version.

**Comment by fballiano** [comment]
As a personal opinion I'd say, instead of running the upgrade script, simply take your project and copy the openmage files on top of it, many many files will be overwritten and that's ok.

then clear your cache and refresh. finish.

notes
- no need to run composer anything
- test locally, there could be incompatibilities since we're talking about years of development

**Comment by addison74** [comment]
When I upgrade to a new version I prefer to delete all the files from the previous version of course with some exceptions and not without creating a full backup. Then I copy the new variant so I make sure there is nothing old left somewhere. 

I don't have the script at hand but it is based on the files contained in a version and which it deletes one by one. Important files, extensions, product images, and more stay untouched. Then the new version is copied in Magento root. It can be done manually but it takes longer to go through each folder and delete the files. For me this method is the safest because a script can give errors and then they can be difficult to identify. 

After that I do a series of tests according to a list to make sure I can access accounts, place orders, search for products and many other facilities for the visitor. It takes about a few days depending on the time I have available. When I'm sure of changes I'll put it into production.

**Comment by luigifab** [comment]
I suggest to use composer, then you can: `composer update`.
If you didn't use it, you need to install.

I don't understand the readme, it seems that it doesn't work to just run: `composer require "openmage/magento-lts":"^19.4.0"`.
The [right way](https://www.openmage.org/magento-lts/install.html) is to create a `composer.json` file in a new empty directory (here for 19.x.x):
```
{
    "require": {
        "aydin-hassan/magento-core-composer-installer": "*",
        "openmage/magento-lts": "^19.4.0"
    },
    "extra": {
        "magento-core-package-type": "magento-source",
        "magento-root-dir": "htdocs"
    }
}
```
Then in the same directory run: `composer install`.

After that you need to copy/paste your files in `htdocs/` (for example: `app/design/frontend/your/design`, `app/skin/frontend/your/design` ...). Same for modules (in `app/code/community` and `app/code/local`), perhaps you can reinstall some of them with composer.

**Comment by MrJarlsburg** [comment]
Thank you so much for your replies.
I downloaded version 19.4.16 archive from https://github.com/OpenMage/magento-lts/tags, replaced all files, emptied the cache.
Site seems to still work, but when I log into the admin panel, it still says OpenMage LTS 19.4.8.
Any ideas? If I read Mage.php it clearly says 'patch' ==> '16'.

**Reply by addison74** [reply]
magento-lts-19.4.16 archive contains only in two files references to version 19.4.8, files that are not run but only for information.

I didn't analyze where the value of the version displayed in the Backend footer is taken from but if you say that you followed all the steps correctly all you have to do is look for this 19.4.8 string in the database and in the OM root file system. If you do not find it it means that it comes from the browser cache.

**Reply by addison74** [reply]
My advice is to enable the OpenMage logs for a while, exceptions and system, plus the PHP log after such an upgrade. There are situations in which you have the impression that there are no errors but they are still unknown. I would take a look periodically in the MySQL and Apache/Nginx logs.

**Reply by fballiano** [reply]
Do you ha e opcache enabled with a very hard refresh policy?

**Comment by MrJarlsburg** [comment, marked as answer]
Sorry. My bad. Version info is correct now.
